github_test.go

```diff
@@ -100,63 +173,75 @@ var _ = Describe("GitHub Client", func() {
 		})
 	})
 
-	Describe("when the source is configured with the deprecated user field", func() {
+	Describe("ListReleases with access token", func() {
 		BeforeEach(func() {
 			source = Source{
-				User:       "some-owner",
-				Repository: "some-repo",
-			}
-
-			server.AppendHandlers(
-				ghttp.CombineHandlers(
-					ghttp.VerifyRequest("GET", "/repos/some-owner/some-repo/releases"),
-					ghttp.RespondWith(200, "[]"),
-				),
-			)
-		})
-
-		It("uses the provided user as the owner", func() {
-			_, err := client.ListReleases()
-			Ω(err).ShouldNot(HaveOccurred())
-		})
-	})
-
-	Describe("ListReleases", func() {
-		BeforeEach(func() {
-			source = Source{
-				Owner:      "concourse",
-				Repository: "concourse",
+				Owner:       "concourse",
+				Repository:  "concourse",
+				AccessToken: "test",
 			}
 		})
-		Context("When list of releases return more then 100 items", func() {
+		Context("List graphql releases", func() {
 			BeforeEach(func() {
 				var result []*github.RepositoryRelease
 				for i := 1; i < 102; i++ {
-					result = append(result, &github.RepositoryRelease{ID: github.Int(i)})
-
+					result = append(result, &github.RepositoryRelease{ID: github.Int64(int64(i))})
 				}
 
 				server.AppendHandlers(
 					ghttp.CombineHandlers(
-						ghttp.VerifyRequest("GET", "/repos/concourse/concourse/releases", "per_page=100"),
-						ghttp.RespondWithJSONEncoded(200, result[:100], http.Header{"Link": []string{`</releases?page=2>; rel="next"`}}),
+						ghttp.VerifyRequest("POST", "/graphql"),
+						ghttp.RespondWith(200, multiPageResp),
 					),
 					ghttp.CombineHandlers(
-						ghttp.VerifyRequest("GET", "/repos/concourse/concourse/releases", "per_page=100&page=2"),
-						ghttp.RespondWithJSONEncoded(200, result[100:]),
+						ghttp.VerifyRequest("POST", "/graphql"),
+						ghttp.RespondWith(200, singlePageResp),
 					),
 				)
 			})
 
 			It("list releases", func() {
 				releases, err := client.ListReleases()
 				Ω(err).ShouldNot(HaveOccurred())
-				Expect(releases).To(HaveLen(101))
+				Expect(releases).To(HaveLen(3))
```

Comment:
Probably a good idea to test the response we get back, i.e.

```
Expect(releases).To(Equal([]*github.RepositoryRelease{{ID: ...}, ...}))
```

since we have no coverage on the ID parsing (or that all the graphql attributes are being mapped properly)

Reply:
Also, we should add a test for an invalid ID (e.g. bad base64 encoding)